libs/core/langchain_core/messages/ai.py

```diff
@@ -35,27 +71,73 @@ class AIMessageChunk(AIMessage, BaseMessageChunk):
     # non-chunk variant.
     type: Literal["AIMessageChunk"] = "AIMessageChunk"  # type: ignore[assignment] # noqa: E501
 
+    tool_call_chunks: Optional[List[ToolCallChunk]] = None
+    """If provided, tool call chunks associated with the message."""
+
     @classmethod
     def get_lc_namespace(cls) -> List[str]:
         """Get the namespace of the langchain object."""
         return ["langchain", "schema", "messages"]
 
+    @root_validator()
+    def init_tool_calls(cls, values: dict) -> dict:
+        if values["tool_calls"] is not None:
+            raise ValueError(
+                "tool_calls cannot be set on AIMessageChunk, it is derived "
+                "from tool_call_chunks."
+            )
+        if not values["tool_call_chunks"]:
```

Comment:
This is a little bit confusing - why not just always overwrite to either empty array or `None`?

Reply:
when summing AIMessageChunks you can get chunks with valid tool calls.

Reply:
nvm got your meaning, updated this.